tests/db_functions/test_datetime.py

```diff
@@ -62,12 +62,34 @@ def truncate(value, kind):
 class DateFunctionTests(TestCase):
 
     def create_model(self, start_datetime, end_datetime):
+
+        name = "None"
+        start_date = None
+        end_date = None
+        start_time = None
+        end_time = None
+        duration = timedelta(0)
+
+        if start_datetime: 
+            name = start_datetime.isoformat()
+            start_date = start_datetime.date()
+            start_time = start_datetime.time()
+
+        if end_datetime: 
+            end_date = end_datetime.date()
+            end_time = end_datetime.time()
+
+        if start_datetime != end_datetime: 
+            duration = end_datetime - start_datetime
+
         return DTModel.objects.create(
-            name=start_datetime.isoformat(),
+            name=name, 
             start_datetime=start_datetime, end_datetime=end_datetime,
-            start_date=start_datetime.date(), end_date=end_datetime.date(),
-            start_time=start_datetime.time(), end_time=end_datetime.time(),
-            duration=(end_datetime - start_datetime),
+            start_date=start_date, 
+            end_date=end_date, 
+            start_time=start_time, 
+            end_time=end_time, 
+            duration=duration, 
```

Comment:
Drop `end_datetime` onto a new line also as the rest of the lines have been changed...